getAndParseWikipediaPOI.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+function getAndParseWikipediaPOI($language, $user_latitude, $user_longitude, $range, $maxPOI, $thumbnailWidth)
+{
+	// ===> Apply limits of Wikipedia API
+	$range = min($range, 10);
+	$maxPOI = min($maxPOI, 500);
+
+	// ===> Call Wikipedia GeoSearch API to get pages around
+	$apidata_request_geosearch = "https://".$language.".wikipedia.org/w/api.php?action=query&list=geosearch&gslimit=".$maxPOI."&gsradius=".($range * 1000)."&gscoord=".$user_latitude."|".$user_longitude."&format=json";
+	if(!($apidata_json_geosearch = @file_get_contents($apidata_request_geosearch)))
+		return array("error" => "Wikipedia API is unreachable");
+
+	// ===> Case their is no POI around at all
+	if(!($apidata_array_geosearch = @json_decode($apidata_json_geosearch, true)['query']['geosearch']))
+		return array();
+
+	$wikipedia_pagesid_list = "";
+
+	// ===> Parse those data in the output array
+	foreach ($apidata_array_geosearch as $currentPOI => $currentPOIdata) {
+		$wikipedia_pagesid_list .= '|'.$currentPOIdata['pageid'];
+		$output_array[$currentPOIdata['pageid']]['latitude'] = $apidata_array_geosearch[$currentPOI]['lat'];
+		$output_array[$currentPOIdata['pageid']]['longitude'] = $apidata_array_geosearch[$currentPOI]['lon'];
+		$output_array[$currentPOIdata['pageid']]['distance'] = $apidata_array_geosearch[$currentPOI]['dist'];
+		$output_array[$currentPOIdata['pageid']]['wikipedia_id'] = $currentPOIdata['pageid'];
+	}
+	$wikipedia_pagesid_list = substr($wikipedia_pagesid_list, 1);
+
+	// ===> Now we got a list of wikipedia pages so we call Wikipedia API again to get infos on those pages
+	$apidata_request_wikipedia_info = "https://".$language.".wikipedia.org/w/api.php?format=json&action=query&prop=pageprops|info|pageimages&inprop=url&pilimit=1000&pithumbsize=".$thumbnailWidth."&pageids=".$wikipedia_pagesid_list;
+	if(!($apidata_json_wikipedia_info = @file_get_contents($apidata_request_wikipedia_info)))
+		return array("error" => "Wikipedia API is unreachable");
+
+	$apidata_array_wikipedia_info = json_decode($apidata_json_wikipedia_info,1);
```

Comment:
Why is it sometimes json_decode(json, true) and sometimes json_decode(json, 1) ?